File: golang/masmaint/model/entity/department.go
package entity

import (
	"database/sql"

	"masmaint/dto"
	"masmaint/core/utils"
)


type Department struct {
	Code string `db:"code"`
	Name string `db:"name"`
	Description sql.NullString `db:"description"`
	ManagerId sql.NullInt64 `db:"manager_id"`
	Location sql.NullString `db:"location"`
	Budget float64 `db:"budget"`
	CreatedAt string `db:"created_at"`
	UpdatedAt string `db:"updated_at"`
}

func NewDepartment() *Department {
	return &Department{}
}

func (e *Department) SetCode(code any) error {
	e.Code = utils.ToString(code)
	return nil
}

func (e *Department) SetName(name any) error {
	e.Name = utils.ToString(name)
	return nil
}

func (e *Department) SetDescription(description any) error {
	if description == nil {
		e.Description.Valid = false
		return nil
	}

	e.Description.String = utils.ToString(description)
	e.Description.Valid = true
	return nil
}

func (e *Department) SetManagerId(managerId any) error {
	if managerId == nil || managerId == "" {
		e.ManagerId.Valid = false
		return nil
	}

	x, err := utils.ToInt64(managerId)
	if err != nil {
		return err
	}
	e.ManagerId.Int64 = x
	e.ManagerId.Valid = true
	return nil
}

func (e *Department) SetLocation(location any) error {
	if location == nil {
		e.Location.Valid = false
		return nil
	}

	e.Location.String = utils.ToString(location)
	e.Location.Valid = true
	return nil
}

func (e *Department) SetBudget(budget any) error {
	x, err := utils.ToFloat64(budget)
	if err != nil {
		return err
	}
	e.Budget = x
	return nil
}

func (e *Department) SetCreatedAt(createdAt any) error {
	e.CreatedAt = utils.ToString(createdAt)
	return nil
}

func (e *Department) SetUpdatedAt(updatedAt any) error {
	e.UpdatedAt = utils.ToString(updatedAt)
	return nil
}


func (e *Department) ToDepartmentDto() dto.DepartmentDto {
	var dDto dto.DepartmentDto

	dDto.Code = e.Code
	dDto.Name = e.Name
	if e.Description.Valid != false {
		dDto.Description = e.Description.String
	}
	if e.ManagerId.Valid != false {
		dDto.ManagerId = e.ManagerId.Int64
	}
	if e.Location.Valid != false {
		dDto.Location = e.Location.String
	}
	dDto.Budget = e.Budget
	dDto.CreatedAt = e.CreatedAt
	dDto.UpdatedAt = e.UpdatedAt

	return dDto
}

File: php/masmaint/src/Application/Models/Entities/Department.php
<?php

declare(strict_types=1);

namespace App\Application\Models\Entities;

use JsonSerializable;

class Department implements JsonSerializable
{
    private string $code;

    private string $name;

    private ?string $description;

    private ?int $managerId;

    private ?string $location;

    private float $budget;

    private string $createdAt;

    private string $updatedAt;

    public function getCode(): string
    {
        return $this->code;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function getManagerId(): ?int
    {
        return $this->managerId;
    }

    public function getLocation(): ?string
    {
        return $this->location;
    }

    public function getBudget(): float
    {
        return $this->budget;
    }

    public function getCreatedAt(): string
    {
        return $this->createdAt;
    }

    public function getUpdatedAt(): string
    {
        return $this->updatedAt;
    }

    public function setCode($code)
    {
        $this->code = $code;
    }

    public function setName($name)
    {
        $this->name = $name;
    }

    public function setDescription($description)
    {
        $this->description = $description;
    }

    public function setManagerId($managerId)
    {
        if ($managerId === null || $managerId === "") {
            $this->managerId = null;
        } else if (filter_var($managerId, FILTER_VALIDATE_INT) !== false) {
            $this->managerId = (int) $managerId;
        } else {
            throw new \InvalidArgumentException("error: setManagerId");
        }
    }

    public function setLocation($location)
    {
        $this->location = $location;
    }

    public function setBudget($budget)
    {
        if ($budget === null || $budget === "") {
            throw new \InvalidArgumentException("error: setBudget");
        } else if (is_numeric($budget)) {
            $this->budget = (float) $budget;
        } else {
            throw new \InvalidArgumentException("error: setBudget");
        }
    }

    public function setCreatedAt($createdAt)
    {
        $this->createdAt = $createdAt;
    }

    public function setUpdatedAt($updatedAt)
    {
        $this->updatedAt = $updatedAt;
    }

    //json_encode()でエンコードされるときに呼ばれる
    #[\ReturnTypeWillChange]
    public function jsonSerialize(): array
    {
        return [
            'code' => $this->code,
            'name' => $this->name,
            'description' => $this->description,
            'manager_id' => $this->managerId,
            'location' => $this->location,
            'budget' => $this->budget,
            'created_at' => $this->createdAt,
            'updated_at' => $this->updatedAt,
        ];
    }
}
